packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -300,16 +300,44 @@ class Scrollable extends StatefulWidget {
   /// ScrollableState? scrollable = Scrollable.maybeOf(context);
   /// ```
   ///
-  /// Calling this method will create a dependency on the closest [Scrollable]
-  /// in the [context], if there is one.
+  /// Calling this method will create a dependency on the [ScrollableState]
+  /// that is returned, if there is one. This is typically the closest
+  /// [Scrollable], but may be a more distant ancestor if [axis] is used to
+  /// target a specific [Scrollable].
+  ///
+  /// Using the optional [Axis] is useful when Scrollables are nested and the
+  /// target [Scrollable] is not the closest instance. When [axis] is provided,
+  /// the nearest enclosing [ScrollableState] in that [Axis] is returned, or
+  /// null if there is none.
   ///
   /// See also:
   ///
   /// * [Scrollable.of], which is similar to this method, but asserts
   ///   if no [Scrollable] ancestor is found.
-  static ScrollableState? maybeOf(BuildContext context) {
-    final _ScrollableScope? widget = context.dependOnInheritedWidgetOfExactType<_ScrollableScope>();
-    return widget?.scrollable;
+  static ScrollableState? maybeOf(BuildContext context, { Axis? axis }) {
+    if (axis == null) {
+      final _ScrollableScope? widget = context.dependOnInheritedWidgetOfExactType<_ScrollableScope>();
+      return widget?.scrollable;
+    }
+    // This is the context that will need to establish the dependency.
+    final BuildContext originalContext = context;
+    ScrollableState? scrollable = context.getInheritedWidgetOfExactType<_ScrollableScope>()?.scrollable;
+    while (scrollable != null) {
+      if (axisDirectionToAxis(scrollable.axisDirection) == axis) {
+        if (context != originalContext) {
+          // Establish the dependency on the correct context.
+          // Get the element.
+          final InheritedElement element = context.getElementForInheritedWidgetOfExactType<_ScrollableScope>()!;
```

Comment:
To consider: I think I would find this code a little cleaner if you remove the `getInheritedWidgetOfExactType` calls altogether and instead replace them with `getElementForInheritedWidgetOfExactType`. From there you can access the scrollable via `element.widget.scrollable` (probably needs a cast). IMO, it makes it a little clearer that the element you depend on is the one you checked the axis on. Right now, it look a little like you look up one inherited widget, check something on it, and then look up another widget to depend on. (sure, in the end you look up the same thing, but it could be made explicit).

Reply:
Maybe simplify further and remove the branching. The code inside the `if` is also correct when context == originalContext, right?

Reply:
Ok! Will do, that would definitely be cleaner